File: code/python/剑指offer55-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None: return 0
        queue = [root]
        deep = 1
        last = root
        while queue:
            cur = queue[0]
            queue = queue[1:]
            if cur.left is not None:
                queue.append(cur.left)
            if cur.right is not None:
                queue.append(cur.right)
            if last==cur:
                if queue:
                    last = queue[-1]
                    deep+=1
        return deep

File: code/java/剑指offer55-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null) return 0;
        int deep = 1;
        List<TreeNode> q = new ArrayList<>();
        q.add(root);
        TreeNode last = root;
        while(q.size()>0){
            TreeNode cur = q.get(0);
            q.remove(0);
            if(cur.left!=null) q.add(cur.left);
            if(cur.right!=null) q.add(cur.right);
            if(last==cur){
                if(q.size()>0){
                    last = q.get(q.size()-1);
                    deep++;
                }
            }
        }
        return deep;
    }
}